tests/lib/ocpp/v201/test_charge_point.cpp

```diff
@@ -340,4 +534,92 @@ TEST_F(ChargePointFixture, K01FR02_CallbacksValidityChecksIfOptionalTransactionE
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), DEFAULT_TX_ID);
+
+    SetChargingProfileRequest req;
+    req.evseId = DEFAULT_EVSE_ID;
+    req.chargingProfile = profile;
+
+    auto set_charging_profile_req =
+        request_to_enhanced_message<SetChargingProfileRequest, MessageType::SetChargingProfile>(req);
+
+    EXPECT_CALL(*smart_charging_handler, validate_profile(testing::_, testing::_));
+    ON_CALL(*smart_charging_handler, validate_profile)
+        .WillByDefault(testing::Return(ProfileValidationResultEnum::Valid));
+    EXPECT_CALL(*smart_charging_handler, add_profile(testing::_, testing::_));
+
+    charge_point->handle_message(set_charging_profile_req);
+}
+
+class ChargePointFixture_InvalidProfiles : public ChargePointFixture,
+                                           public ::testing::WithParamInterface<ProfileValidationResultEnum> {};
+
+INSTANTIATE_TEST_SUITE_P(
+    ChargePointOnlyAddValidProfilesTests, ChargePointFixture_InvalidProfiles,
+    testing::Values(ProfileValidationResultEnum::EvseDoesNotExist, ProfileValidationResultEnum::InvalidProfileType,
+                    ProfileValidationResultEnum::TxProfileMissingTransactionId,
+                    ProfileValidationResultEnum::TxProfileEvseIdNotGreaterThanZero,
+                    ProfileValidationResultEnum::TxProfileTransactionNotOnEvse,
+                    ProfileValidationResultEnum::TxProfileEvseHasNoActiveTransaction,
+                    ProfileValidationResultEnum::TxProfileConflictingStackLevel,
+                    ProfileValidationResultEnum::ChargingProfileNoChargingSchedulePeriods,
+                    ProfileValidationResultEnum::ChargingProfileFirstStartScheduleIsNotZero,
+                    ProfileValidationResultEnum::ChargingProfileMissingRequiredStartSchedule,
+                    ProfileValidationResultEnum::ChargingProfileExtraneousStartSchedule,
+                    ProfileValidationResultEnum::ChargingScheduleChargingRateUnitUnsupported,
+                    ProfileValidationResultEnum::ChargingSchedulePeriodsOutOfOrder,
+                    ProfileValidationResultEnum::ChargingSchedulePeriodInvalidPhaseToUse,
+                    ProfileValidationResultEnum::ChargingSchedulePeriodUnsupportedNumberPhases,
+                    ProfileValidationResultEnum::ChargingSchedulePeriodExtraneousPhaseValues,
+                    ProfileValidationResultEnum::ChargingSchedulePeriodPhaseToUseACPhaseSwitchingUnsupported,
+                    ProfileValidationResultEnum::ChargingStationMaxProfileCannotBeRelative,
+                    ProfileValidationResultEnum::ChargingStationMaxProfileEvseIdGreaterThanZero,
+                    ProfileValidationResultEnum::DuplicateTxDefaultProfileFound,
+                    ProfileValidationResultEnum::DuplicateProfileValidityPeriod));
+
+TEST_P(ChargePointFixture_InvalidProfiles, K01_SetChargingProfileRequest_OnlyAddsValidProfiles) {
```

Comment:
Maybe instead of _OnlyAddsValidProfiles, it should be _DoesNotAddInvalidProfiles ?